### PR TITLE
3482 - update bundle doc

### PR DESCRIPTION
**Pull request by wyli**
Signed-off-by: Wenqi Li <wenqil@nvidia.com>

part of #3482 

### Description
preview here https://github.com/wyli/MONAI/blob/3482-bundle-doc/docs/source/config_syntax.md

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: tests/testing_data/metadata.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "schema": "https://github.com/Project-MONAI/MONAI-extra-test-data/releases/download/0.8.1/meta_schema_202203171008.json",
+    "schema": "https://github.com/Project-MONAI/MONAI-extra-test-data/releases/download/0.8.1/meta_schema_20220324.json",
```
Comment (Nic-Ma) [inline tests/testing_data/metadata.json:2]: I checked the schema file, seems you forget to update the key to verify:
```
"authorship": {
    "description": "state author(s) of the model package.",
    "type": "string"
}
```

Thanks.

File: docs/source/config_syntax.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
+# MONAI Bundle Configuration
```
Comment (Nic-Ma) [inline docs/source/config_syntax.md:1]: I think the `OmegaConf` documentation can be a good reference to introduce the regular operations and syntax:
https://omegaconf.readthedocs.io/en/2.1_branch/

Thanks.

File: docs/source/config_syntax.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+```
+
+_Description:_ A reference to another configuration value defined at `preprocessing#transforms#keys`.
+where `#` indicates a sub-structure of this configuration file.
+
+```json
+"@preprocessing#1"
+```
+
+_Description:_ `1` is interpreted as an integer, which is used to index the `preprocessing` sub-structure.
```
Comment (Nic-Ma) [inline docs/source/config_syntax.md:89]: Suggest to describe 2 points: "list" and "index from 0", maybe:
which is used to index (count from `0`) the `preprocessing` sub-structure list.

Thanks.

File: docs/source/config_syntax.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+Python expressions. The imported module `resnet18` will be available as a global variable
+to the other configuration sections. This is to simplify the use of external modules in the configuration.
+
+### 3. `%` to interpolate with plain texts
+
+```json
+"%demo_config.json#demo_net#in_channels"
+```
+
+_Description:_ A macro to replace the current value with the texts at `demo_net#in_channels` in the
```
Comment (Nic-Ma) [inline docs/source/config_syntax.md:119]: Maybe: A macro to replace the current **config text** value with the texts...

File: docs/source/config_syntax.md [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+    "_target_": "my.python.module.Class",
+    "args1": "string",
+    "args2": 42}
+}
+```
+
+_Description:_ This dictionary defines an object with a reference name `demo_name`, with an instantiable type
+specified at `_target_` and with input arguments `args1` and `args2`.
+This dictionary will be instantiated as a Pytorch object at runtime.
+`_target_` is a required key by monai bundle syntax for the Python object name.
```
Comment (Nic-Ma) [inline docs/source/config_syntax.md:136]: Please add a blank line before this line to make it a new paragraph.

Thanks.

File: docs/source/config_syntax.md [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+- it describes workflow at a relatively high level and allows for different low-level implementations.
+- learning paradigms at a higher level such as federated learning and AutoML can be decoupled from the component details.
+
+Content:
+
+- [A basic example](#a-basic-example)
+- [Syntax examples explained](#syntax-examples-explained)
+  - [`@` to interpolate with Python objects](#1--to-interpolate-with-python-objects)
+  - [`$` to evaluate as Python expressions](#2--to-evaluate-as-python-expressions)
+  - [`%` to interpolate with plain texts](#3--to-interpolate-with-plain-texts)
```
[the remark this answers is not in the document]
Comment (Nic-Ma) [reply docs/source/config_syntax.md:17]: Maybe `to textually replace with other config texts`.

Thanks.

File: monai/bundle/config_parser.py [hunk 7]
```diff
@@ -211,12 +211,16 @@ def get_parsed_content(self, id: str = "", **kwargs):
                 Use digits indexing from "0" for list or other strings for dict.
                 For example: ``"xform#5"``, ``"net#channels"``. ``""`` indicates the entire ``self.config``.
             kwargs: additional keyword arguments to be passed to ``_resolve_one_item``.
-                Currently support ``reset`` (for parse), ``instantiate`` and ``eval_expr``. All defaulting to True.
+                Currently support ``lazy`` (whether to retain the current config cache, default to `False`),
```
[the remark this answers is not in the document]
Comment (wyli) [reply monai/bundle/config_parser.py:214]: yes, during writing the demo I found `config.get_parsed_content(..., reset=False)` is confusing, it's not clear which part is not 'reset'.

[the remark this answers is not in the document]
Comment (Nic-Ma) [reply monai/bundle/config_parser.py:214]: Maybe change the `reset` arg of parser to `lazy` directly? I am also not very sure about it...

Thanks.

**Review by Nic-Ma — APPROVED** [review]
Thanks for the quick update.
Looks good to me.

Thanks.

**Pull request merged**